pkg/rtc/mediatrack.go

```diff
@@ -138,86 +139,88 @@ func (t *MediaTrack) AddReceiver(receiver *webrtc.RTPReceiver, track *webrtc.Tra
 		}
 	})
 
-	isNew := false
 	t.lock.Lock()
-	if t.Receiver() == nil {
-		isNew = true
-
-		wr := sfu.NewWebRTCReceiver(
+	mime := strings.ToLower(track.Codec().MimeType)
+	t.params.Logger.Debugw("AddReceiver", "mime", track.Codec().MimeType)
+	wr := t.MediaTrackReceiver.Receiver(mime)
+	if wr == nil {
+		var priority int
+		for idx, c := range t.params.TrackInfo.Codecs {
```

Comment:
Is the order of `TrackInfo.Codecs` determined by the client? during addTrackRequest?

also curious on the ordering of negotiating with subscribers. since the published tracks are coming in at different times, is it possible for this to happen?
1. server receives h.264
2. server creates the downtrack, and completes negotiation for h.264
3. server receives av1

Is this possible?

Reply:
> Is the order of TrackInfo.Codecs determined by the client? during addTrackRequest?

yeah, client control that

> also curious on the ordering of negotiating with subscribers. since the published tracks are coming in at different times, is it possible for this to happen?...

No, server side will wait all codecs of a track(declared in addTrackRequest ) has been received, then transit it from `pending` from `published`, before that, subscriber can't see the track.